### PR TITLE
Adding a microbenchmark for side input iterables.

### PR DESCRIPTION
**Pull request by pabloem**
r: @aaltay 

This measures time spent fetching from side inputs. Intended for use in side input metrics measures.

### CODE REVIEW

**Comment by pabloem** [comment]
Adding a passing precommit, as I plan to update num of iterations and num of elements, but shouldn't affect the test performance of the job.

https://builds.apache.org/job/beam_PreCommit_Python_GradleBuild/1154/console

**Comment by pabloem** [comment]
Python precommit is passing

File: sdks/python/apache_beam/tools/sideinput_microbenchmark.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+
+def long_generator(value, elements):
+  for _ in range(elements):
+    yield value
+
+
+def run_benchmark(num_runs=50, input_per_source=4000, num_sources=4):
+  print("Number of runs:", num_runs)
+  print("Input size:", num_sources*input_per_source)
```
Comment (aaltay) [inline sdks/python/apache_beam/tools/sideinput_microbenchmark.py]: Do we run lint on these files ?

I would have thought that there will be a space around `*`.

Comment (pabloem) [reply sdks/python/apache_beam/tools/sideinput_microbenchmark.py]: I've fixed this. And lint does run on these files, so I guess we don't have the checkf or that style.

**Comment by pabloem** [comment]
Python tests have passed. Merging 
https://builds.apache.org/job/beam_PreCommit_Python_GradleBuild/1196/console

**Pull request merged**